unepwcmc/ProtectedPlanet
# Refresh search

## Description

Adds search functionality for areas and full site (CMS).

## Notes

Make sure your db is up to date and migrated.
Reindex elasticsearch like the following:

```
> Search::Index.delete
> reload!
> Search::Index.create
> Search::Index.create_cms_fragments
```

⚠️ Please note that indexing needs improvement, so search results might not work exactly as expected sometimes.

## Main things to test

* Search from the home page should search across WDPA and OECM
* Search from the WDPA page should redirect to search with WDPA filter pre-selected
* Search from the OECM page should redirect to search with OECM filter pre-selected
* Search filters are working ok
* Search geo-types (Region, Country, Individual Areas) are working ok and matching the search term
* Search results images are loading (please note some might show as blank, this is something to be addressed later)
* Search from top nav should redirect to full site search
* Full site search categories should filter results correctly
* Clicking on a result card should redirect to the related resource
